test/TesterInternal/OrleansRuntime/Streams/ObjectPoolTests.cs

```diff
@@ -72,5 +75,38 @@ public void Alloc10Free1Test()
             // only 1 at a time was ever allocated, but pool was pre-populated with 10, so max allocated should be 10
             Assert.AreEqual(10, accumulator.MaxAllocated);
         }
+
+        [Fact, TestCategory("BVT"), TestCategory("Streaming")]
+        public void ReuseResourceTest()
+        {
+            var accumulator = new Accumulator();
+            IObjectPool<TestPooledResource> pool = new ObjectPool<TestPooledResource>(() => new TestPooledResource(accumulator), 10);
+
+            for (int i = 0; i < 10; i++)
+            {
+                // Allocate 10 items
+                var resources = Enumerable.Range(0, 10).Select(v => pool.Allocate()).ToList();
+
+                resources.Reverse(); // reversing alloca to maintain order when disposed
+
+                // free 10
+                resources.ForEach(r => r.Dispose());
+
+                // pool was pre-populated with 10, so max allocated should be 10
+                Assert.AreEqual(10, accumulator.MaxAllocated);
+
+                // Allocate and free 10 items
+                for (int ii = 0; ii < 20; ii++)
+                {
+                    TestPooledResource resource = pool.Allocate();
+                    Assert.AreEqual((i*21)+(ii+1), resource.AllocationCount);
+                    resource.Dispose();
+                }
```

Comment:
lol, in Math, in Argentina at least, we use "j" instead of "ii" for the second dimension. It's fine, I just found it surprising :smile: